app/views/videos/show.html.haml

```diff
@@ -11,7 +11,7 @@
         %p= @video.description
         .actions
           %a.btn.btn-primary(href="") Watch Now
-          %a.btn.btn-default(href="") + My Queue
+          = link_to "+ My Queue", queue_items_path(video_id: @video.id), method: :post, class: "btn btn-default"
```

Comment:
I think you should allow this link only when the Video is not in the queue already.